adserver/utils.py

```diff
@@ -87,14 +97,18 @@ def get_client_ip(request):
     Gets the real IP based on a request object.
 
     Checks if ``request.ip_address`` is present from a Middleware
-    (eg. ``RealIPAddressMiddleware``) and returns that.
-    If that is not set, return the value from ``REMOTE_ADDR``.
+    (eg. ``CloudflareIpAddressMiddleware``) and returns that.
+    If that is not set, return the value from ``REMOTE_ADDR`` and raise a warning.
     """
-    ip = getattr(request, "ip_address", None)
-    if ip:
-        return ip
+    if hasattr(request, "ip_address"):
+        return request.ip_address
 
-    return request.META.get("REMOTE_ADDR", "")
+    log.warning(
+        "No IP address set by middleware (see CloudflareIpAddressMiddleware). "
+        "Consider enabling an IP address middleware so IPs can be used for ad targeting."
+    )
```

Comment:
Nice warning 👍 

Though it will still use the `REMOTE_ADDR` to target, so maybe this message is a bit off?

Reply:
I think it's a pretty important warning to raise. Without some middleware `REMOTE_USER` is almost certain to be wrong. It's probably just a load balancer's IP.